app/Http/Controllers/Api/RestoreDatabaseController.php

```
@@ -0,0 +1,333 @@
+            $backups = $pgBackRestService->listBackups();
+
+            return response()->json([
+                'backups' => $backups,
+            ], 200);
+        } catch (\Exception $e) {
+            return response()->json([
+                'message' => 'Failed to list backups.',
+                'error' => $e->getMessage(),
+            ], 500);
+        }
+    }
```

PHPMD (2.15.0)

The method list_backups() has an NPath complexity of 256. The configured NPath complexity threshold is 200. (undefined)

(NPathComplexity)